report/sections/lessons-learned-perspective/operation-maintenance-evolution.tex

```diff
@@ -0,0 +1,29 @@
+\subsection{Operation, maintenance and evolution}
+
+\subsubsection{Initial refactoring}
+
+At the start of the project, we hosted a large collaborative session, using Live Share \cite{tool:live-share} to let everyone see the same context of the workspace of one developer, allowing us to co-edit and co-debug the same code.
+In retrospect, perhaps it would have been better to commit a basic boilerplate Gin app, and then have each developer slowly commit more changes.
+
+Having attempted a 1:1-translation, we ended up with one large file. This is where we started separating responsibilites, moving them to smaller files, as well as moving the front-end to a separate repo.
+
+
+\subsubsection{Replacing database management systems}
+
+At some point, we switched from SQLite to SQL Server. This could be done almost seamlessly due to the abstraction of our repository controllers, as described in section \ref{section:system-design}.
+This abstraction was introduced in PR \#21, particularly \href{https://github.com/Devops-2022-Group-R/itu-minitwit/pull/21/commits/96d0ea601453ff9d3efe52c30b2435c6308f120b}{\color{blue}this commit}.
```

Comment:
We should avoid having links that can only be visited via PDF on a computer. Make this a source instead and provide the short commit hash in the text.

Reply:
Also, the linked commit question is not the one we should be referring to. Refer to the one where we replace the database provider.